packages/api-v2/src/ws-client/request.ts

```diff
@@ -0,0 +1,71 @@
+import {
+    TSocketResponseData,
+    TSocketResponse,
+    TSocketEndpointNames,
+    TSocketRequestPayload,
+    TSocketSubscribableEndpointNames,
+} from '../../types';
+
+const REQ_TIMEOUT = 20000;
+
+// sequence number for requests
+let reqSeqNumber = 0;
+
+/**
+ * responsible for sending request over given WS and thats it,
+ * no handling of reconnections, no state, nothing, just send
+ * even request seq number is outside of its scope (reason being, that req_seq needs to be also used by the subscriptions)
+ */
+function request<T extends TSocketSubscribableEndpointNames>(
+    ws: WebSocket,
+    name: TSocketEndpointNames,
+    payload: TSocketRequestPayload<T>['payload']
+): Promise<TSocketResponse<T>> {
+    const req_id = ++reqSeqNumber;
+
+    const promise: Promise<TSocketResponseData<T>> = new Promise((resolve, reject) => {
```

Comment:
What happens if the connection is closed when this is called? Does it throw?

Reply:
yes, WebSocket api will throw an exception if its closed,

Reply:
Is that what currently happens? It doesn't queue?